scripts/quests/windurst/Rock_Racketeer.lua

```diff
@@ -0,0 +1,230 @@
+}
+
+quest.sections =
+{
+    {
+        -- QUEST AVAILABLE.  Talk to Nanaa Mihgo, get the KI.
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getFameLevel(xi.quest.fame_area.WINDURST) >= 3 and
+                player:hasCompletedQuest(xi.quest.log_id.WINDURST, xi.quest.id.windurst.MIHGO_S_AMIGO) and
+                player:getCurrentMission(xi.mission.log_id.WINDURST) ~= xi.mission.id.windurst.LOST_FOR_WORDS and
+                quest:getMustZone(player) == false
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Nanaa_Mihgo'] = quest:progressEvent(93),
+
+            onEventFinish =
+            {
+                [93] = function(player, csid, option, npc)
+                    if
+                        option == 1 and
+                        npcUtil.giveKeyItem(player, xi.ki.SHARP_GRAY_STONE)
+                    then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Talk to Ardea in Bastok Markets
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and
+            player:hasKeyItem(xi.ki.SHARP_GRAY_STONE)
+        end,
+
+        [xi.zone.BASTOK_MARKETS] =
+        {
+            ['Ardea'] = quest:progressEvent(261),
+
+            onEventFinish =
+            {
+                [261] = function(player, csid, option, npc)
+                    if
+                        option ~= 1 and
+                        npcUtil.giveCurrency(player, "gil", 10)
+                    then
+                        player:delKeyItem(xi.ki.SHARP_GRAY_STONE)
+                        player:messageSpecial(bastokID.text.KEYITEM_OBTAINED + 1, xi.ki.SHARP_GRAY_STONE)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Nanaa_Mihgo'] = quest:progressEvent(95),
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and
+            not player:hasKeyItem(xi.ki.SHARP_GRAY_STONE) and
+            quest:getVar(player, "Progress") < 1
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Nanaa_Mihgo'] =
+            {
+                onTrigger = function (player, npc)
+                    return quest:progressEvent(98)
+                end
+            },
+
+            onEventFinish =
+            {
+                [98] = function(player, csid, option, npc)
+                    return quest:setVar(player, "Progress", 1)
+                end,
+            },
+        },
+    },
+
+    -- Talk to Varun in Windurst Woods
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and
+            not player:hasKeyItem(xi.ki.SHARP_GRAY_STONE) and
+            quest:getVar(player, "Progress") == 1
```

Comment:
Think the norm here is Prog instead of Progress

Reply:
Had "Progress" from somewhere on LSB so just went with it.  Either way is good with me.

Reply:
Updated.